### ISSUE TITLE
Unable to add integer array to document

### ISSUE THREAD

**Issue by ardevd** [appwrite/appwrite#794]
## 🐛 Bug Report

After adding an array value to an existing document type, I'm unable to set the array value of any document.

## To Reproduce

Create a new database table (document type?). 

Add a integer array field to it.

Find an exisitng document, and try to set the integer array value. A "Failed to Update Document" error message is shown in the UI. I also notice that even if I make the field "required", no documents actually show the value in request responses either.

## Expected behavior

The document should update correctly when adding array values.

**Comment by eldadfux** [comment]
@ardevd please post your collection structure, and the payload errors your API is returning from the devtools network tab.

**Comment by ardevd** [comment]
```json
{
    "$id": "5f8b6153a7828",
    "$collection": 0,
    "$permissions": {
        "read": [
            "*"
        ],
        "write": []
    },
    "name": "sotainfo",
    "dateCreated": 1602969939,
    "dateUpdated": 1609575817,
    "structure": true,
    "rules": [
        {
            "$id": "5f8b61efc2dd1",
            "$collection": "rules",
            "$permissions": {
                "read": [
                    "*"
                ],
                "write": []
            },
            "label": "Component",
            "key": "component",
            "type": "text",
            "required": true,
            "array": false,
            "default": ""
        },
        {
            "$id": "5f8b622719f6b",
            "$collection": "rules",
            "$permissions": {
                "read": [
                    "*"
                ],
                "write": []
            },
            "label": "Release Date",
            "key": "release_date",
            "type": "numeric",
            "required": true,
            "array": false,
            "default": 0
        },
        {
            "$id": "5f8b625690627",
            "$collection": "rules",
            "$permissions": {
                "read": [
                    "*"
                ],
                "write": []
            },
            "label": "Version",
            "key": "version",
            "type": "text",
            "required": true,
            "array": false,
            "default": ""
        },
        {
            "$id": "5f8b6256930c9",
            "$collection": "rules",
            "$permissions": {
                "read": [
                    "*"
                ],
                "write": []
            },
            "label": "Vehicle Model",
            "key": "vehicle",
            "type": "text",
            "required": true,
            "array": false,
            "default": "X590"
        },
        {
            "$id": "5f8b626b7cd84",
            "$collection": "rules",
            "$permissions": {
                "read": [
                    "*"
                ],
                "write": []
            },
            "label": "Summary",
            "key": "summary",
            "type": "text",
            "required": true,
            "array": false,
            "default": ""
        },
        {
            "$id": "5f8b628b22e23",
            "$collection": "rules",
            "$permissions": {
                "read": [
                    "*"
                ],
                "write": []
            },
            "label": "URL",
            "key": "link",
            "type": "url",
            "required": true,
            "array": false
        },
        {
            "$id": "5f8ed77ebe512",
            "$collection": "rules",
            "$permissions": {
                "read": [
                    "*"
                ],
                "write": []
            },
            "label": "Signature",
            "key": "signature",
            "type": "text",
            "required": true,
            "array": false,
            "default": ""
        },
        {
            "$id": "5ff02b916c877",
            "$collection": "rules",
            "$permissions": {
                "read": [
                    "*"
                ],
                "write": []
            },
            "label": "Model Years",
            "key": "model_years",
            "type": "numeric",
            "required": true,
            "array": true,
            "default": 2019
            }
      ]
}
```

Where do I find the devtools network tab containing the requested errors?

**Comment by eldadfux** [comment]
@ardevd this is an example:

![image](https://user-images.githubusercontent.com/1297371/104412060-40672180-5574-11eb-8556-792bd25b467d.png)

**Comment by ardevd** [comment]
```json
{"message":"Bad structure. Invalid document structure: Property \"model_years\" must be an array","code":400,"version":"0.6.2"}
```

**Comment by ardevd** [comment]
Am I misintepreting the UI somehow? The UI seens to let you "add" integer values to the field, which to me means that it should add the supplied integers to the array. Yet the error message seems to indicate its not being treated as array values at all.

**Comment by eldadfux** [comment]
@ardevd to get the full picture and verify if this is a bug or not we'll need the request payload as well

**Comment by ardevd** [comment]
```json
{"data":{"component":"IMC","release_date":1596412800000,"version":"test","vehicle":"X590","summary":"test","link":"https://<redacted>/Rwnng6fMP/download","signature":"AQ3L3gwwRgIhAKQj51NbzW7Dtu2yJ32uKHF9N8NVWmsMhLDxM+0yY2l4AiEAxibeCrcG20nHBJok96WFpqya2USEHpOtTt2risnTC2g=","model_years":2012},"read":["*"],"write":[]}
```

**Comment by eldadfux** [comment]
Your `model_years` is expecting any array according to your collection, but your json has an integer value:

```json
{
	"data": {
		"component": "IMC",
		"release_date": 1596412800000,
		"version": "test",
		"vehicle": "X590",
		"summary": "test",
		"link": "https://<redacted>/Rwnng6fMP/download",
		"signature": "AQ3L3gwwRgIhAKQj51NbzW7Dtu2yJ32uKHF9N8NVWmsMhLDxM+0yY2l4AiEAxibeCrcG20nHBJok96WFpqya2USEHpOtTt2risnTC2g=",
		"model_years": 2012
	},
	"read": ["*"],
	"write": []
}
```

Try and change it something like this:

```json
{
	"data": {
		"component": "IMC",
		"release_date": 1596412800000,
		"version": "test",
		"vehicle": "X590",
		"summary": "test",
		"link": "https://<redacted>/Rwnng6fMP/download",
		"signature": "AQ3L3gwwRgIhAKQj51NbzW7Dtu2yJ32uKHF9N8NVWmsMhLDxM+0yY2l4AiEAxibeCrcG20nHBJok96WFpqya2USEHpOtTt2risnTC2g=",
		"model_years": [2012]
	},
	"read": ["*"],
	"write": []
}
```

**Comment by ardevd** [comment]
Right, but its supposed to be a numerical array. Seems to me like there's a bug, either in the UI or the request generator.

![screenshot_2021-01-13-205444](https://user-images.githubusercontent.com/366855/104503688-ac33a380-55e1-11eb-8144-b7e3ed451a24.png)

**Comment by ardevd** [comment]
> 
> Try and change it something like this:
> 
> ```json
> {
> 	"data": {
> 		"component": "IMC",
> 		"release_date": 1596412800000,
> 		"version": "test",
> 		"vehicle": "X590",
> 		"summary": "test",
> 		"link": "https://<redacted>/Rwnng6fMP/download",
> 		"signature": "AQ3L3gwwRgIhAKQj51NbzW7Dtu2yJ32uKHF9N8NVWmsMhLDxM+0yY2l4AiEAxibeCrcG20nHBJok96WFpqya2USEHpOtTt2risnTC2g=",
> 		"model_years": [2012]
> 	},
> 	"read": ["*"],
> 	"write": []
> }
> ```

Yep, that works fine. Any chance for a proper fix though?

**Comment by ardevd** [comment]
Also, after updating the record manually, if I try to add an additional integer array value to the same record, it also fails, and the network requests shows that it doesnt include the other array value. 

```json
{"data":{"component":"IMC","release_date":1596412800000,"version":"S20B1_20.30.2-578440","vehicle":"X590","summary":"Identical to S20B_20.30.2-578440","link":"https://nc.redbird.no/index.php/s/7xz2TiRwnng6fMP/download","signature":"AQ3L3gwwRgIhAKQj51NbzW7Dtu2yJ32uKHF9N8NVWmsMhLDxM+0yY2l4AiEAxibeCrcG20nHBJok96WFpqya2USEHpOtTt2risnTC2g=","model_years":2019},"read":["*"],"write":[]}
```

It should be 
```json
{"data":{"component":"IMC","release_date":1596412800000,"version":"S20B1_20.30.2-578440","vehicle":"X590","summary":"Identical to S20B_20.30.2-578440","link":"https://nc.redbird.no/index.php/s/7xz2TiRwnng6fMP/download","signature":"AQ3L3gwwRgIhAKQj51NbzW7Dtu2yJ32uKHF9N8NVWmsMhLDxM+0yY2l4AiEAxibeCrcG20nHBJok96WFpqya2USEHpOtTt2risnTC2g=","model_years":[2012,2019]},"read":["*"],"write":[]}
```

**Comment by eldadfux** [comment]
OK, I can see the problem in the UI, can you see the same issue in the API?

**Comment by ardevd** [comment]
What do you mean? If I modify and fix the API query and have it pass array values it works fine. The problem seems to be the malformed requests generated by the UI.

**Comment by eldadfux** [comment]
Yes, that what I wanted to know, thanks, I'll mark this as a bug and the team will work on a fix.

**Comment by ardevd** [comment]
This is still broken in 0.7. Setting array values through the UI doesnt work at all and I need to do it through cURL every time I want to edit my documents.

**Comment by eldadfux** [comment]
We're working on some refactoring around this issues, probably aiming this to 0.9 or 0.10 releases

**Comment by ardevd** [comment]
Ok, cool. No need for me to come in and fix array specific issues then. I'll deal with curl for now ;)

**Comment by TorstenDittmann** [comment]
Fixed the UI bug in #1070
